src/features/home/api/helpers/mapOffersDataAndModules.ts

```diff
@@ -0,0 +1,59 @@
+import { SearchResponse } from '@algolia/client-search'
+import { flatten, uniqBy } from 'lodash'
+import { UseQueryResult } from 'react-query'
+
+import { ModuleData, OfferModuleParamsInfo } from 'features/home/types'
+import { AlgoliaHit } from 'libs/algolia'
+import { filterOfferHit } from 'libs/algolia/fetchAlgolia/transformOfferHit'
+import { Offer } from 'shared/offer/types'
+
+interface MapProps {
+  results: UseQueryResult<SearchResponse<Offer>[], unknown>
+  modulesParams: (OfferModuleParamsInfo | undefined)[]
+  transformHits: (hit: AlgoliaHit) => AlgoliaHit
+}
+
+export const mapOffersDataAndModules = ({ results, modulesParams, transformHits }: MapProps) => {
+  const { data } = results
+
+  if (data) {
+    let moduleIterator = 0
+
+    const offersModulesData = modulesParams.map((module) => {
+      if (module) {
+        if (module.nbParams === 0) {
+          const value = {
+            hits: [],
+            nbHits: 0,
+            moduleId: module.moduleId,
+          }
+
+          return value as ModuleData
```

Comment:
On est obligé de forcer le typage ?
On peut pas juste le spécifier en créant l'objet ?
```suggestion
          const value: ModuleData = {
            hits: [],
            nbHits: 0,
            moduleId: module.moduleId,
          }

          return value
```